sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs

```diff
@@ -287,194 +437,404 @@ public class EventProcessorClient : EventProcessorBase<PartitionContext>, IAsync
+                            .ConfigureAwait(false);
+
+                        InstanceOwnership.Clear();
+
+                        // We need to wait until all tasks have stopped before making the load balancing task null.  If we did it sooner, we
+                        // would have a race condition where the user could update the processing handlers while some pumps are still running.
+
+                        ActiveLoadBalancingTask = null;
+                    }
                 }
-                catch (Exception)
+                finally
                 {
-                    // TODO: delegate the exception handling to an Exception Callback.
-                    // Maybe we should just surface the exception.
+                    RunningTaskSemaphore.Release();
                 }
             }
         }
 
         /// <summary>
-        ///   Responsible for processing events received from the Event Hubs service.
+        ///   Stops the event processor.  In case it isn't running, nothing happens.
         /// </summary>
         ///
-        /// <param name="partitionEvent">The partition event to be processed.</param>
-        /// <param name="context">The context in which the associated partition is being processed.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        public virtual void StopProcessing(CancellationToken cancellationToken = default) => StopProcessingAsync(cancellationToken).GetAwaiter().GetResult();
+
+        /// <summary>
+        ///   Determines whether the specified <see cref="System.Object" /> is equal to this instance.
+        /// </summary>
+        ///
+        /// <param name="obj">The <see cref="System.Object" /> to compare with this instance.</param>
+        ///
+        /// <returns><c>true</c> if the specified <see cref="System.Object" /> is equal to this instance; otherwise, <c>false</c>.</returns>
+        ///
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override bool Equals(object obj) => base.Equals(obj);
+
+        /// <summary>
+        ///   Returns a hash code for this instance.
+        /// </summary>
+        ///
+        /// <returns>A hash code for this instance, suitable for use in hashing algorithms and data structures like a hash table.</returns>
+        ///
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override int GetHashCode() => base.GetHashCode();
+
+        /// <summary>
+        ///   Converts the instance to string representation.
+        /// </summary>
+        ///
+        /// <returns>A <see cref="System.String" /> that represents this instance.</returns>
+        ///
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override string ToString() => base.ToString();
```

Comment:
Should we return the Identifier here?   Or something like `$"Event Processor: { Identifier }`?